### ISSUE TITLE
app.svelte

### ISSUE THREAD

**Issue by david-plugge** [sveltejs/kit#8939]
### Describe the problem

The entry point to a sveltekit application is the `app.html` file inside `src`. Because this is just plain html we are not able to customize anything during runtime and have to use `transformPageChunk` to make these changes.

### Describe the proposed solution

Since we are all about svelte, lets make it a svelte component!

```svelte
<script lang="ts">
	import { RequestEvent } from '@sveltejs/kit';

    export let event: RequestEvent;
</script>

<!DOCTYPE html>
<html lang={event.locals.lang}>
	<head>
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
		<link rel="icon" href="/favicon.png" />

		<slot name="head"/>
	</head>
	<body class="theme-{event.locals.theme}">
		<div>
			<slot name="body"/>
		</div>
	</body>
</html>
``` 

### Importance

nice to have

**Comment by david-plugge** [comment]
Basically just the ability to set stuff like the document language and a theme class at runtime instead of using `transformPageChunk` since we don´t have something like `<svelte:html>` or `<svelte:root>`.

No big deal, just in case this hasn't been considered yet.

**Comment by Rich-Harris** [comment]
The use of an `.html` file is deliberate — we want as few moving parts as possible when rendering pages so that there's one less place errors can happen, one less layer of indirection, etc. If we had an`app.svelte` then the issue tracker would be full of people asking why `app.svelte` isn't reactive. (Which it can't be, of course: even if Svelte components could contain `<!doctype html>`, and the DOM allowed you to _provide_ `document.documentElement` rather than merely working inside it, in the browser you don't have access to `event`.)

Some frameworks _do_ allow you to declare a component for the entire document, but it's really an elaborate fiction. 

And I've seen cases where having precise control over the page template is important (for example, people creating embeddable widgets, where the consumer needs to be able to target `<!-- MY_SIGNIFICANT_COMMENT -->` which is the sort of thing Svelte omits when rendering), and any abstraction like this would get in the way.

In short, unless there's something that's impossible to achieve with `transformPageChunk` (which exists for precisely the illustrated use case), it's vanishingly unlikely that we'd do this.
